### PR TITLE
initial forum channel support

### PR DESCRIPTION
**Pull request by TopiSenpai**
https://github.com/discord/discord-api-docs/pull/4761

### CODE REVIEW

File: discord/channel.go [hunk 1]
```diff
@@ -27,8 +27,52 @@ const (
...
 	ChannelTypeGuildStageVoice
 	ChannelTypeGuildDirectory
+	ChannelTypeGuildForum
 )
 
+type ChannelFlags int
+
+const (
+	ChannelFlagsNone   ChannelFlags = 0
+	ChannelFlagsPinned ChannelFlags = 1 << iota
```
Comment (Skye-31) [inline discord/channel.go:37]: Perhaps rename this to ChannelFlagsThreadPinned to make it clearer?

File: discord/channel.go [hunk 2]
```diff
@@ -909,6 +1042,38 @@ func ApplyGuildIDToChannel(channel GuildChannel, guildID snowflake.Snowflake) Gu
 		c.guildID = guildID
 		return c
 	default:
-		panic("unknown channel type")
+		panic(fmt.Sprintf("invalid channel type: %d", channel.Type()))
```
Comment (Skye-31) [inline discord/channel.go:1045]: Is panicking the best idea here? It's always possible discord adds a new type and this should not break production bots. It should probably warn instead.

**Review by Skye-31 — CHANGES_REQUESTED** [review]

File: discord/channel.go (hunk 1 above)
Comment (TopiSenpai) [reply discord/channel.go:37]: it's documented as pinned
maybe they plan to do other stuff with it so I disagree

File: discord/channel.go (hunk 2 above)
Comment (TopiSenpai) [reply discord/channel.go:1045]: hm yeah prob

File: discord/channel.go (hunk 1 above)
Comment (Skye-31) [reply discord/channel.go:37]: image
This is an example of it, I don't think they do

Comment (TopiSenpai) [reply discord/channel.go:37]: I'd keep it as close to the docs as possible to avoid breaking changes later 
¯\_(ツ)_/¯

Comment (Skye-31) [reply discord/channel.go:37]: The name of an enum is not breaking if discord changes it

Comment (TopiSenpai) [reply discord/channel.go:37]: if they change it so you can pin normal channels it would be

Comment (Skye-31) [reply discord/channel.go:37]: The description of the flag specifically indicates it's for threads
> this thread is pinned to the top of its parent forum channel

Comment (TopiSenpai) [reply discord/channel.go:37]: then we can add an description 
keeping the naming consistent with the docs is still the way to go

**Comment by TopiSenpai** [comment]
this should be tested first before merging

File: discord/channel.go (hunk 1 above)
Comment (Skye-31) [inline discord/channel.go:37]: To make it consistent with other flag naming in the library
```suggestion
	ChannelFlagNone   ChannelFlags = 0
	ChannelFlagPinned ChannelFlags = 1 << iota
```

**Review by Skye-31 — APPROVED** [review]

**Comment by TopiSenpai** [comment]
there are some new good changes on forum thread creation https://github.com/discord/discord-api-docs/pull/4850

**Review by Skye-31 — APPROVED** [review]

**Pull request merged**